### PR TITLE
[runtime] swift_getObjectType() must ignore KVO's artificial subclasses.

### PR DESCRIPTION
**Pull request by gparker42**
Fixes SR-4363 and <rdar://problem/31275541>

### CODE REVIEW

**Comment by gparker42** [comment]
@swift-ci please test

File: stdlib/public/runtime/SwiftObject.mm [hunk 1]
```diff
@@ -79,6 +79,34 @@ static uintptr_t computeISAMask() {
...
+///
+/// The object pointer may be a tagged pointer, but cannot be null.
+const Metadata *swift::swift_getObjectType(HeapObject *object) {
+  auto classAsMetadata = _swift_getClass(object);
+
+  // Walk up the superclass chain skipping over artifical Swift classes.
+  // If we find a non-Swift class use the result of [object class] instead.
+
+  while (classAsMetadata && classAsMetadata->isTypeMetadata()) {
+    if (!classAsMetadata->isArtificialSubclass())
```
Comment (jrose-apple) [inline stdlib/public/runtime/SwiftObject.mm]: Is this interesting to do on non-ObjC platforms?

Comment (jckarter) [inline stdlib/public/runtime/SwiftObject.mm]: It's unnecessary, but harmless.

Comment (jckarter) [inline stdlib/public/runtime/SwiftObject.mm]: To that point, we could probably move the `#if SWIFT_OBJC_INTEROP` block up here.

**Review by jckarter — APPROVED** [review]
Looks good for 4.0, though as Jordan pointed out we could skip the walk through artificial subclasses on non-`OBJC_INTEROP` runtimes.

**Comment by gparker42** [comment]
@swift-ci please test

**Comment by gparker42** [comment]
@swift-ci please test

**Comment by gparker42** [comment]
Mac test failure looked legit. I haven't tried to track it down yet.

**Comment by gparker42** [comment]
@swift-ci please test

**Comment by gparker42** [comment]
@swift-ci please test

**Comment by gparker42** [comment]
@swift-ci please test

**Pull request merged**